boa_engine/src/object/mod.rs

```diff
@@ -1951,20 +2346,21 @@ where
 
 /// Builder for creating native function objects
 #[derive(Debug)]
-pub struct FunctionObjectBuilder<'ctx, 'host> {
-    context: &'ctx mut Context<'host>,
+pub struct FunctionObjectBuilder {
+    realm: Realm,
```

Comment:
Can we extract this to its own PR? I saw that a considerable part of the PR changes were because of this, and I think if this wasn't part of the PR it'll make reviews easier.

Reply:
Yeah, I think I can separate them out. I think they were part of the initial commits and mostly just kept them because they made sense. I'll try to get that completed tonight.

Reply:
Ah that's right. The changes to `FunctionObjectBuilder` and `ObjectInitializer` stem from constructing the `Temporal.Now` object on `Temporal`. We only have access to `Realm` from `Init`, so the change was made to accomodate that. We could remove the initialization for `Temporal.Now` from this PR and submit it as a different PR as a follow up.

Reply:
Hm, then should this use what `Intl` uses? It just references empty intrinsic objects, then the objects are initialized in `Realm` construction.

Reply:
https://github.com/boa-dev/boa/blob/3c15e59e6717e018f826bf0b915faedff994d646/boa_engine/src/builtins/intl/mod.rs#L52-L56

Reply:
The Temporal.Now object:

 - is an [ordinary object](https://tc39.es/ecma262/#ordinary-object).
 - has a [[Prototype]] internal slot whose value is [%Object.prototype%](https://tc39.es/ecma262/#sec-properties-of-the-object-prototype-object).
 - is not a [function object](https://tc39.es/ecma262/#function-object).
 - does not have a [[Construct]] internal method; it cannot be used as a [constructor](https://tc39.es/ecma262/#constructor) with the new operator.
 - does not have a [[Call]] internal method; it cannot be invoked as a function.
 
 😕 Spec says it doesn't have a constructor

Reply:
Yes, but you can also use `BuiltInBuilder` directly to initialize a custom intrinsic that is not a constructor, just like [`Math`](https://github.com/boa-dev/boa/blob/3c15e59e6717e018f826bf0b915faedff994d646/boa_engine/src/builtins/math/mod.rs#L35-L46)